src/bin/server/config.rs

```diff
@@ -1,12 +1,12 @@
-use std::{io, net::SocketAddr};
+use std::io;
 
 use drop::crypto::{key::exchange, sign};
 use snafu::{ResultExt, Snafu};
 
 #[derive(serde::Deserialize, serde::Serialize)]
 pub struct ConfigAddresses {
-    pub node: SocketAddr,
-    pub rpc: SocketAddr,
+    pub node: String,
```

Comment:
is there no type for a node address? Like URL or something?

Reply:
I need smth that can resolve to a `SocketAddr`, using [`ToSocketAddrs`](https://docs.rs/tokio/1.13.0/tokio/net/trait.ToSocketAddrs.html). As you can see in the doc, it either needs an IP, a `String` or `(String, u16)`. Having the first one isn't fit for docker and cumbersome for the user. The last one is also a bit weird to write in toml (`node = ["example.com", 80]`) and don't add any safety to have it separated IMO.

It's not an URL either: there is not location of resources, only of the host, and the protocol is always AT2 specific. So the only relevant parts would be the host and the port, which is what we have here

Reply:
As you might have imagined, I didn't check out the documentation...